### ISSUE TITLE
API Plantnet

### ISSUE THREAD

**Issue by orovellotti** [PnX-SI/gn_mobile_occtax#234]
Les Bureau d'études (UPGE) sont très demandeur de l'intégration d'une IA dans l'outil.

https://my.plantnet.org/

Plantnet a une API qui a apparemment est facile d'utilisation.
Eneo l'a intégré dans son outil, ils peuvent nous aider.

Qu'est ce que vous en dites ?

**Comment by camillemonchicourt** [comment]
Oui pourquoi pas si c'est un module indépendant et optionnel, pour ne pas rendre Occtax-mobile dépendant de Plantnet.
A voir les coûts et conditions.

**Comment by DonovanMaillard** [comment]
@orovellotti as-tu pu explorer davantage le sujet depuis ta proposition ?

**Comment by camillemonchicourt** [comment]
J'ai discuté récemment avec les équipes de PlantNet et ils étaient très motivés pour nous fournir des clés de leur API gratuitement.

**Comment by DonovanMaillard** [comment]
Un avis du CBNA notamment sur la question @jpm-cbna ? Sur la fiabilité des déterminations notamment ? 

Il y a aussi Obsidentify qui fait faune et flore , je ne sais pas dans quelles mesures l'API est ouverte ou non. Assez bluffante sur les papillons de nuit mais laisse passer des aberrations dans le lot...

**Comment by camillemonchicourt** [comment]
PlantNet je pense que ça fait consensus pour une première aide à l'identification.
OK, [Obsidentify](https://observation.org/apps/obsidentify/) semble être l'outil derrière l'excellent [Observation.org](https://observation.org/).
Il y a aussi un bon moteur d'identification photo dans iNaturalist.

L'avantage de PlantNet est qu'il est pensé pour être intégré dans des outils tiers comme GeoNature et qu'on peut obtenir des clés API.

**Comment by DonovanMaillard** [comment]
Oui, mais ça ne fait que les plantes et occtax propose la saisie de tous les taxon, c'est le sens de ma réflexion. Ca implique d'avoir une IA disponible uniquement pour une partie des saisies. Mieux que rien, mais on peut peut être faire encore mieux. 

Je ne sais pas si le NIA qui est derrière Obsentify, observation.org etc, est accessible via une API ouverte à d'autres applis : https://www.naturalis.nl/en/ai

**Comment by orovellotti** [comment]
Je peux tout refaire en Vibe code et ajouter PantNet ?

**Comment by DonovanMaillard** [comment]
un petit truc en vibe code pour tester une ou plusieurs manières d'implémenter la fonctionnalité, pourquoi pas. Refaire occtax-mobile avec tout ce que ça comporte, je n'essaierais pas ^^

**Comment by jpm-cbna** [comment]
> Un avis du CBNA notamment sur la question [@jpm-cbna](https://github.com/jpm-cbna) ? Sur la fiabilité des déterminations notamment ?

Plantnet doit être utilisé comme une aide à la détermination. C'est très efficace pour faire gagner du temps sur les déterminations en permettant de cibler un genre directement. A titre personnel, je m'en sers souvent pour les déterminations de mes photos. Je me sert ensuite de Flora Gallica pour vérifier/affiner la détermination.

Par contre, il faudrait pouvoir intégrer Plantnet dans OccTax de façon à pouvoir aussi ajouter facilement d'autre API d'identification. Et ces API devraient pouvoir être activé/désactivé facilement sans casser le fonctionnement d'OccTax.
